jwst/stpipe/step.py

```diff
@@ -714,28 +714,53 @@ def get_config_from_reference(cls, dataset, observatory=None, disable=None):
         step_parameters : configobj
             The parameters as retrieved from CRDS. If there is an issue, log as such
             and return an empty config obj.
+
+        Notes
+        -----
+        The CRDS lookup normally uses `META.INSTRUMENT.NAME` as the primary key, to distinguish
+        references at the instrument `imap` level. However, the step parameter reference files
+        are all grouped in a pseudo-instrument "CALPARS". As such, the instrument name is set to
+        "CALPARS". If an rmap requires knowledge of the original instrument, this is located
```

Comment:
I assume (and from what I've seen in the new code above) that it's not the case that the instrument name is simply being reset to "calpars", because how then would a call to CRDS find any instrument-level reference files? But in reality it's being given multiple instrument names, one of which is "calpars" and the other is the original instrument (e.g. "miri", "nircam", ...)? Is that correct? If so, maybe the Notes and/or code comments could make that a little more clear.